src/apps/intel_mp/intel_mp.lua

```diff
@@ -1278,16 +1296,23 @@ function Intel82599:rxdmapackets ()
 end
 
 function Intel82599:init_queue_stats (frame)
+   local function keys(t)
+      local ret = {}
+      for k,_ in pairs(t) do table.insert(ret, k) end
+      table.sort(ret)
+      return ret
+   end
    local perqregs = {
       rxdrops = "QPRDC",
       rxpackets = "QPRC",
-      txpackets = "QPTC",
       rxbytes = "QBRC64",
       txbytes = "QBTC64",
+      txpackets = "QPTC",
    }
    self.queue_stats = {}
    for i=0,15 do
-      for k,v in pairs(perqregs) do
+      for _,k in ipairs(keys(perqregs)) do
```

Comment:
I think this keys/ipairs thing is probably unneeded -- it makes the code longer for a function only called during init

Reply:
I need to insert the registers in order, to decouple rx registers and tx registers. This allows me to update based on rxcounter (updates from 1..3 perqregs) and txcounter (updated 4 and 5).